src/aiida/orm/nodes/data/list.py

```diff
@@ -81,15 +82,15 @@ def remove(self, value):
             self.set_list(data)
         return item
 
-    def pop(self, **kwargs):
+    def pop(self, index: int = -1) -> Any:
```

Comment:
This seems like potentially breaking change. Can you maybe add type ignore for now and then submit a separate PR where we can discuss more (it would be good to not block this PR to unblock pre-commit failures)

Reply:
For reference, I am copying the mypy error:

```python
src/aiida/orm/nodes/data/list.py:84: error: Signature of "pop" incompatible with supertype "MutableSequence"  [override]
src/aiida/orm/nodes/data/list.py:84: note:      Superclass:
src/aiida/orm/nodes/data/list.py:84: note:          def pop(self, index: int = ...) -> Any
src/aiida/orm/nodes/data/list.py:84: note:      Subclass:
src/aiida/orm/nodes/data/list.py:84: note:          def pop(self, **kwargs: Any) -> Any
src/aiida/orm/nodes/data/list.py:92: error: Signature of "index" incompatible with supertype "Sequence"  [override]
src/aiida/orm/nodes/data/list.py:92: note:      Superclass:
src/aiida/orm/nodes/data/list.py:92: note:          def index(self, value: Any, start: int = ..., stop: int = ...) -> int
src/aiida/orm/nodes/data/list.py:92: note:      Subclass:
src/aiida/orm/nodes/data/list.py:92: note:          def index(self, value: Any) -> Any
```

Reply:
Moved to PR #6635